server/filestore.go

```diff
@@ -5615,6 +5585,28 @@ func (fs *fileStore) expireMsgs() {
 		}
 	}
 
+	// THW is unordered, so must sort by sequence and must not be holding the lock.
+	if len(ttlSdm) > 0 {
+		fs.mu.Unlock()
+		for subj, es := range ttlSdm {
+			// THW is unordered, so must sort by sequence first.
+			slices.SortFunc(es, func(a, b SDMBySubj) int {
```

Comment:
I realise there probably shouldn't be multiple `SDMBySubj`s with the same sequence, but it's still important to return 0 on equality here. The `slices.SortFunc` function relies on strict weak ordering and it can tight-loop if two entries with the same sequence did end up here and you return 1, either by accident or otherwise.

Reply:
Done